components/controls/RadioToggle.tsx

```diff
@@ -18,52 +18,58 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 import * as classNames from 'classnames';
+import * as _ from 'lodash';
 import * as React from 'react';
 import './RadioToggle.css';
 import Tooltip from './Tooltip';
 
-interface Option {
+interface Option<T = string> {
   disabled?: boolean;
   label: string;
   tooltip?: string;
-  value: string;
+  value: T;
 }
 
-interface Props {
+interface Props<T = string> {
   className?: string;
   name: string;
-  onCheck: (value: string) => void;
-  options: Option[];
-  value?: string;
+  onCheck: (value: T | undefined) => void;
+  options: Option<T>[];
+  value?: T;
 }
 
-export default class RadioToggle extends React.PureComponent<Props> {
+export default class RadioToggle<T = string> extends React.PureComponent<Props<T>> {
   static defaultProps = {
     disabled: false,
     value: null
   };
 
   handleChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
     const newValue = e.currentTarget.value;
-    this.props.onCheck(newValue);
+    // Let assum that if we can't found the option we have unselect
```

Comment:
`We assume that if we can't find the option, it was unselected`.
Although I'm not sure this is an actual use-case. I would call `onCheck` only if the option is found.
Let's see what Greg thinks.